### PR TITLE
replace HAPI FHIR / madie-fhir-service dependency with cql-library-service

### PR DESCRIPTION
**Pull request by nmorasb**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-5401](https://jira.cms.gov/browse/MAT-5401)
(Optional) Related Tickets:
[MAT-5445](https://jira.cms.gov/browse/MAT-5445)

### Summary
Replace HAPI FHIR / madie-fhir-service dependency with cql-library-service.  
Also includes some small changes to enable use of QDM libraries.

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/cql_translator/MadieLibrarySourceProvider.java [hunk 1]
```diff
@@ -11,17 +11,39 @@
...
+        .libraryType(threadLocalValue.get().getLibraryType())
+        .version(threadLocalValue.get().getVersion())
+        .line(threadLocalValue.get().getLine())
+        .comment(threadLocalValue.get().getComment())
+        .build();
+  }
 
-  public static void setFhirServicesService(MadieFhirServices madieFhirServices) {
-    MadieLibrarySourceProvider.madieFhirServices = madieFhirServices;
+  public static void setFhirServicesService(CqlLibraryService cqlLibraryService) {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/mat/cql_elm_translation/cql_translator/MadieLibrarySourceProvider.java]: Nothing major, but Is there a better name for this method, it's trippy and FHIR could be omitted. Maybe setSourceProvider or setSourceProviderService

**Review by RohitKandimalla — DISMISSED** [review]

File: src/main/java/gov/cms/mat/cql_elm_translation/cql_translator/MadieLibrarySourceProvider.java (hunk 1 above)
Comment (nmorasb) [reply src/main/java/gov/cms/mat/cql_elm_translation/cql_translator/MadieLibrarySourceProvider.java]: oh - totally right! I missed this when I was using the IDE refactor -> rename functions!

**Review by RohitKandimalla — APPROVED** [review]
